items/managed.js

```diff
@@ -181,6 +181,55 @@ class Item {
 
         return false;
     }
+   
+    /**
+     * Calculates the toggled state of this Item. For Items like Color and 
+     * Dimmer, getStateAs(OnOffType) is used and the toggle calculated off 
+     * of that.
+     * @returns the toggled state (e.g. 'OFF' if the Item is 'ON')
+     * @throws error if the Item is uninitialized or is a type that doesn't make sense to toggle
+     */
+     #getToggleState() {
+        if(this.isUninitialized) {
+            throw Error('Cannot toggle uninitialized Items');
+        }
+        switch (this.type) {
+            case 'PlayerItem' :
+                return this.state == 'PAUSE' ? 'PLAY' : 'PAUSE';
+            case 'ContactItem' :
+                return this.state == 'OPEN' ? 'CLOSED' : 'OPEN';
+            default: {
+                const oldState = this.rawItem.getStateAs(OnOffType);
+                if(oldState) {
+                    return oldState.toString() == 'ON' ? 'OFF' : 'ON';
+                }
+                else {
+                    throw Error('Toggle not supported for items of type ' + this.type);
+                }
+            }
+        }
+    }
+
+    /**
+     * Sends a command to flip the Item's state (e.g. if it is 'ON' an 'OFF' 
+     * command is sent).
+     * @throws error if the Item is uninitialized or a type that cannot be toggled or commanded
+     */
+    toggleCommand() {
+        if(this.type == 'ContactItem'){
+            throw Error('Cannot command Contact Items');
+        }
+        this.sendCommand(this.#getToggleState());
+    }
+
+    /**
+     * Posts an update to flip the Item's state (e.g. if it is 'ON' an 'OFF'
+     * update is posted).
+     * @throws error if the Item is uninitialized or a type that cannot be toggled
+     */
+    toggleUpdate() {
```

Comment:
```suggestion
    postToggleUpdate() {
```
wdyt?  I'm thinking we should try and be consistant since we use `postUpdate` elsewhere

Reply:
Same as above. I'll get a PR submitted later today with these two changes.

Reply:
Done